src/nsls2api/services/pass_service.py

```diff
@@ -121,11 +121,33 @@ async def get_saf_from_proposal(
     return saf_list
 
 
-async def get_commissioning_proposals_by_year(year: int):
+async def get_commissioning_proposals_by_year(year: int, facility: FacilityName = FacilityName.nsls2) -> Optional[list[PassProposal]]:
+
+    pass_facility = await facility_service.pass_id_for_facility(facility)
+    if not pass_facility:
+        error_message: str = f"Facility {facility} does not have a PASS ID."
+        logger.error(error_message)
+        raise PassException(error_message)
+    
     # The PASS ID for commissioning proposals is 300005
-    url = f"{base_url}Proposal/GetProposalsByType/{api_key}/NSLS-II/{year}/300005/NULL"
-    proposals = await _call_pass_webservice(url)
-    return proposals
+    url = f"{base_url}/Proposal/GetProposalsByType/{api_key}/{pass_facility}/{year}/300005/NULL"
+
+    try:
+        pass_commissioning_proposals = await _call_pass_webservice(url)
+        commissioning_proposal_list = []
+        if pass_commissioning_proposals and len(pass_commissioning_proposals) > 0:
+            for commissioning_proposal in pass_commissioning_proposals:
+                commissioning_proposal_list.append(PassProposal(**commissioning_proposal))
+    except ValidationError as error:
+        error_message = f"Error validating commissioning proposal data recevied from PASS for year {str(year)} at {facility} facility."
+        logger.error(error_message)
+        raise PassException(error_message) from error
+    except Exception as error:
+        error_message = f"Error retrieving commissioning proposal information from PASS for year {str(year)} at {facility} facility."
+        logger.exception(error_message)
+        raise PassException(error_message) from error
```

Comment:
This should probably be two separate try-catch blocks

1 for fetching proposals with PassException

1 within the for loop for validating each proposal.

Unless the entire call should fall if one bad proposal is returned?